lms/djangoapps/courseware/courses.py

```diff
@@ -533,26 +535,87 @@ def get_course_assignments(course_key, user, include_access=False):
-                subsection_key, title, url, due, contains_gated_content, complete, past_due, assignment_type
-            ))
+            if due and graded:
+                contains_gated_content = include_access and block_data.get_xblock_field(
+                    subsection_key, 'contains_gated_content', False)
+                title = block_data.get_xblock_field(subsection_key, 'display_name', _('Assignment'))
+
+                assignment_type = block_data.get_xblock_field(subsection_key, 'format', None)
+
+                url = None
+                start = block_data.get_xblock_field(subsection_key, 'start')
+                assignment_released = not start or start < now
+                if assignment_released:
+                    url = reverse('jump_to', args=[course_key, subsection_key])
+
+                complete = is_block_structure_complete_for_assignments(block_data, subsection_key)
+                past_due = not complete and due < now
+                assignments.append(_Assignment(
+                    subsection_key, title, url, due, contains_gated_content, complete, past_due, assignment_type, None
+                ))
+
+            # Load all dates for ORA blocks as separate assignments
+            descendents = block_data.get_children(subsection_key)
+            while descendents:
+                descendent = descendents.pop()
+                descendents.extend(block_data.get_children(descendent))
+                if block_data.get_xblock_field(descendent, 'category', None) == 'openassessment':
+                    graded = block_data.get_xblock_field(descendent, 'graded', False)
+                    has_score = block_data.get_xblock_field(descendent, 'has_score', False)
+                    weight = block_data.get_xblock_field(descendent, 'weight', 1)
+                    if not (graded and has_score and (weight is None or weight > 0)):
+                        continue
+
+                    all_assessments = [{
+                        'name': 'submission',
+                        'due': block_data.get_xblock_field(descendent, 'submission_due'),
+                        'start': block_data.get_xblock_field(descendent, 'submission_start'),
+                        'required': True
+                    }]
+                    valid_assessments = block_data.get_xblock_field(descendent, 'valid_assessments', [])
+
+                    if valid_assessments:
+                        all_assessments.extend(valid_assessments)
+
+                    assignment_type = block_data.get_xblock_field(descendent, 'format', None)
+                    complete = is_block_structure_complete_for_assignments(block_data, descendent)
+
+                    block_title = block_data.get_xblock_field(descendent, 'title', _('Open Response Assessment'))
+
+                    for assessment in all_assessments:
+                        due = parse_date(assessment['due']).replace(tzinfo=pytz.UTC) if assessment['due'] else None
```

Comment:
Now that I'm trying to read this with as much caution as possible, should all these gets in the assessment be moved to `.get`s? i.e.  `assessment['due']` -> `assessment.get('due')`

Reply:
Yeah, think I got all that.